users/views.py

```diff
@@ -1,3 +1,76 @@
-from django.shortcuts import render
+import json
+import re
+import bcrypt
+import jwt
 
-# Create your views here.
+from django.http      import JsonResponse, HttpResponse
+from django.views     import View
+from django.db.models import Q
+
+from users.models     import User
+from my_settings      import SECRET_KEY, JWT_ALGORITHM
+from users.utils      import LoginConfirm, is_valid, \
+                             PW_REGEX, EMAIL_REGEX, NICKNAME_REGEX
+
+class SignupView(View): 
+    def post(self,request):
+        try:
+            data = json.loads(request.body)
+
+            if not is_valid(data['email'],EMAIL_REGEX):
+                return JsonResponse({"MESSAGE":"INVALID_EMAIL_FORMAT"},status=400)
+            
+            if not is_valid(data['password'],PW_REGEX):
+                return JsonResponse({"MESSAGE":"INVALID_PW_FORMAT"},status=400)
+                
+            if not is_valid(data['nickname'],NICKNAME_REGEX):
+                return JsonResponse({"MESSAGE":"INVALID_NICKNAME_FORMAT"},status=400)
+        
+            if User.objects.filter(
+                Q(email    = data['email'])|
+                Q(nickname = data['nickname'])
+                ).exists():
+            
+                return JsonResponse({"MESSAGE":"USER_ALREADY_EXISTS"},status=400)
+                
+            hashed_password = bcrypt.hashpw(data['password'].encode('utf-8'),bcrypt.gensalt())
+            
+            User.objects.create(
+                nickname = data['nickname'],
+                email    = data["email"],
+                password = hashed_password.decode()
+            )
+            
+            return HttpResponse(status=201)
+
+        except KeyError:
+            return JsonResponse({'MESSAGE':"KEY_ERROR"},status=400)
+        
+        except json.JSONDecodeError:
+            return JsonResponse({"MESSAGE":"INVALID_DATA"},status=400)
+    
+class SigninView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+            user = User.objects.get(email=data['email'])
+            
+            if not bcrypt.checkpw(data['password'].encode(),user.password.encode()):
+                return JsonResponse({"MESSAGE":"INVALID_USER"},status=400) 
+            
+            access_token = jwt.encode({"id":user.id},SECRET_KEY,algorithm=JWT_ALGORITHM)
+            
+            return JsonResponse({
+                'MESSAGE'  : "SUCCESS",
+                "NICKNAME" : user.nickname,
+                "TOKEN"    : access_token.decode()
+                },status=201)
```

Comment:
한줄로 충분할 것 같습니다.
* acess_token을 반환 받았다는 의미에는 요청에 대해서 응답이 성공했음을 내포하고 있기 때문에 message : success는 불필요해 보입니다.
* 현재 서버에서 반환해주는 token이 어떤 토큰인지 명확히 해줄 필요가 있어 보입니다. 나중에는 refresh_token, csrf_token등 많은 값들을 관리해야될 상황이 올 수 있기 때문입니다~

```suggestion
            return JsonResponse({
                "nick_name" : user.nickname,
                "acess_token"    : access_token.decode()
                },status=201)
```

Reply:
넵 !